### PR TITLE
feat: Gradle executor docker image for multiple Java versions

### PR DESCRIPTION
**Pull request by exu**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: build/agent/docker-bake.hcl [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
+    "linux/amd64",
+  ]
+}
+
+
+target "jdk18" {
   inherits = ["docker-metadata-action"]
   context = "./"
-  dockerfile = "build/agent/Dockerfile"
+  dockerfile = "build/agent/Dockerfile.jdk17"
```
Comment (vsukhin) [inline build/agent/docker-bake.hcl:31]: jdk18?

File: build/agent/docker-bake-old.hcl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// docker-bake.hcl
```
Comment (vsukhin) [inline build/agent/docker-bake-old.hcl:1]: Do we need to keep it?

File: build/agent/Dockerfile.jdk18 [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+FROM golang:1.17 as builder
+
+WORKDIR /build
+COPY . .
+ENV CGO_ENABLED=0 
+ENV GOOS=linux
+
+RUN cd cmd/agent;go build -o /runner -mod mod -a .
+
+FROM gradle:7.4.2-jdk17
```
Comment (vsukhin) [inline build/agent/Dockerfile.jdk18:11]: jdk17?

File: .github/workflows/docker-multi-build.yaml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            type=semver,pattern={{version}}-jdk18
+            type=semver,pattern={{major}}.{{minor}}-jdk18
+            type=sha
+      - name: Build JDK18 Gradle executor
+        uses: docker/bake-action@v1
+        with:
+          files: |
+            build/agent/docker-bake.hcl
+            ${{ steps.meta_gradle_executor_jdk18.outputs.bake-file }}
+          targets: jdk11
```
Comment (vsukhin) [inline .github/workflows/docker-multi-build.yaml:93]: jdk11?

File: .github/workflows/docker-multi-build.yaml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            type=semver,pattern={{version}}-jdk17
+            type=semver,pattern={{major}}.{{minor}}-jdk17
+            type=sha
+      - name: Build JDK17 Gradle executor
+        uses: docker/bake-action@v1
+        with:
+          files: |
+            build/agent/docker-bake.hcl
+            ${{ steps.meta_gradle_executor_jdk17.outputs.bake-file }}
+          targets: jdk11
```
Comment (vsukhin) [inline .github/workflows/docker-multi-build.yaml:72]: jdk17?

**Review by vsukhin — COMMENTED** [review]
a few questions

File: .github/workflows/docker-multi-build.yaml [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Docker meta for Gradle executor JDK11
+        id: meta_gradle_executor_jdk11
+        uses: docker/metadata-action@v3
+        with:
+          # TODO remove -dev after tests
```
[the remark this answers is not in the document]
Comment (exu) [reply .github/workflows/docker-multi-build.yaml:34]: Yeah it should be a draft PR sorry :)

**Review by vsukhin — APPROVED** [review]
looks good

File: build/agent/docker-bake.hcl [hunk 7]
```diff
@@ -1,10 +1,41 @@
-// docker-bake.hcl
 target "docker-metadata-action" {}
 
-target "build" {
+group "default" {
+    targets = ["jdk11","jdk17","jdk18"]
```
Comment (vsukhin) [inline build/agent/docker-bake.hcl:4]: jdk8?

File: .github/workflows/docker-build.yaml [hunk 8]
```diff
@@ -1,57 +1,166 @@
...
-    tags: 
-      - "[0-9]+.[0-9]+.[0-9]+"
-      - "[0-9]+.[0-9]+.[0-9]+-*"
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-beta*"
 
 jobs:
   docker:
     runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (exu) [reply .github/workflows/docker-build.yaml:12]: I've had hope that you'll comment with something like this to limit repetitions :D

**Pull request merged**